src/features/events_list/atoms/eventSortingConfig.ts

```diff
@@ -0,0 +1,46 @@
+import { action, atom } from '@reatom/framework';
+import { AppFeature } from '~core/app/types';
+import { configRepo } from '~core/config';
+import type { EventsListFeatureConfig } from '../types';
+import type { SortByMCDAScoreConfig } from '~utils/mcda_sort/sortByMCDAScore';
+
+const DEFAULT_SORT_CONFIG: EventSortConfig = { order: 'none', config: null };
+
+export type EventSortConfig = {
+  order: 'asc' | 'desc' | 'none';
+  config: MCDASortConfig | SinglePropertySortConfig | null;
+};
+
+export type MCDASortConfig = {
+  type: 'mcda';
+  mcdaConfig: SortByMCDAScoreConfig;
+};
+
+export type SinglePropertySortConfig = {
+  type: 'singleProperty';
+  propertyName: string;
+};
+
+function getEventSortConfig(): EventSortConfig {
+  const eventsListFeature: EventsListFeatureConfig | null =
+    typeof configRepo.get().features[AppFeature.EVENTS_LIST] === 'object'
+      ? (configRepo.get().features[AppFeature.EVENTS_LIST] as EventsListFeatureConfig)
+      : null;
+  const initialSortConfig = eventsListFeature?.initialSort
+    ? eventsListFeature.initialSort
+    : DEFAULT_SORT_CONFIG;
+  if (initialSortConfig.order !== 'asc' && initialSortConfig.order !== 'desc') {
+    initialSortConfig.order = 'none';
```

Comment:
Here you overriding original sort that came from API. Its better to store original value and config and return new object

Reply:
It would be weird to store one value but return a completely different one. The original is stored within app config anyway, this atom (now, after this new refactor) stores a valid sort order value (in case of undefined or some unknown value just defaulting to none)

Reply:
fixed